lib/src/model/package_graph.dart

```diff
@@ -704,16 +704,25 @@ class PackageGraph {
 
   @override
   String toString() {
-    final divider = '=========================================================';
-    final buffer =
-        StringBuffer('PackageGraph built from ${defaultPackage.name}');
+    const divider = '=========================================================';
+    final buffer = StringBuffer('PackageGraph built from ');
+    buffer.writeln(defaultPackageName);
     buffer.writeln(divider);
     buffer.writeln();
-    for (final name in packageMap.keys) {
-      final package = packageMap[name];
-      buffer.writeln('Package $name documented at ${package.documentedWhere} '
-          'with libraries: ${package.allLibraries}');
-    }
+
+    packageMap.forEach((name, package) {
+      buffer.write('Package ');
```

Comment:
Is this truly that much faster?  The readability problems this introduces just don't seem worth it to me.

Reply:
This one had the biggest win in my benchmarking actually, plus I wanted to add the logic of not including "with libraries" if `allLibraries` was empty anyway, so abstracting out the few other pieces wasn't too much extra.

Reply:
Writing a quick benchmark with just a few packages:

Old(RunTime): 59.896142788691904 us.
New(RunTime): 38.88160734073369 us.

Reply:
I'm mostly concerned about overall performance of the app and overoptimizing.   While it may be true that this is measurably faster when calling `toString()`, how often do we do that in a typical run?  It's more likely that if I'm looking at this function I'm looking to figure out what `toString()` is giving me in a debugger and in that case readability is more important.